### PR TITLE
Support -Xno-optimized-callable-references kotlinc flag

### PR DESCRIPTION
**Pull request by Bencodes**
https://github.com/JetBrains/kotlin/blob/68fdeaf865f279cb7a75eaec7e9f4e6cd0646707/compiler/testData/cli/jvm/extraHelp.out#L77-L78

### CODE REVIEW

**Comment by Bencodes** [comment]
Rebased this one

File: kotlin/internal/opts.bzl [hunk 1]
```diff
@@ -49,6 +49,13 @@ _KOPTS = {
         ),
         type = attr.bool,
     ),
+    "x_no_optimized_callable_references": struct(
+        args = dict(
+            default = False,
+            doc = "Do not use optimized callable reference superclasses available from 1.4.",
```
Comment (restingbull) [inline kotlin/internal/opts.bzl:55]: nit: "Do not use optimized callable reference superclasses. Available from 1.4."

**Review by restingbull — CHANGES_REQUESTED** [review]
Add a check in (_kotlin_toolchain_impl)[ https://github.com/bazelbuild/rules_kotlin/blob/master/kotlin/internal/toolchains.bzl#L49] on the version number is greater than 1.4

**Comment by Bencodes** [comment]
It would probably make more sense to do the validation in `_kotlinc_options_provider_to_flags` so that we can also validate the target level opts that aren't being consumed directly by the toolchain.

**Comment by restingbull** [comment]
> It would probably make more sense to do the validation in `_kotlinc_options_provider_to_flags` so that we can also validate the target level opts that aren't being consumed directly by the toolchain.

Reasonable, considering https://github.com/bazelbuild/rules_kotlin/pull/418.

File: kotlin/internal/jvm/compile.bzl [hunk 2]
```diff
@@ -162,7 +162,7 @@ def _adjust_resources_path(path, resource_strip_prefix):
         return _adjust_resources_path_by_default_prefixes(path)
 
 def _merge_kt_jvm_info(module_name, providers):
-    language_versions = {p.language_version: True for p in providers if p.language_verison}
+    language_versions = {p.language_version: True for p in providers if p.language_version}
```
Comment (Bencodes) [inline kotlin/internal/jvm/compile.bzl:165]: There seems to have been a typo here

**Comment by Bencodes** [comment]
Updated with some basic validation

**Review by restingbull — APPROVED** [review]

**Comment by restingbull** [comment]
Is it me, or the kotlinc flags getting longer?

**Comment by Bencodes** [comment]
Rebased this after all the kotlinc merges.

**Comment by restingbull** [comment]
Ah, thanks. Meant to rebase the pr and got distracted.

**Pull request merged**